pkg/controllers/jobset_controller.go

```diff
@@ -411,6 +402,28 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 	return errors.Join(finalErrs...)
 }
 
+func (r *JobSetReconciler) createJobWithBackOff(ctx context.Context, js *jobset.JobSet, job *batchv1.Job) error {
+	// 1, 2, 4, 8, 16, 32 seconds.
+	backoff := wait.Backoff{
+		Steps:    6,
+		Duration: 1 * time.Second,
+		Factor:   2.0,
+		Jitter:   0.2,
+	}
+	// Set jobset controller as owner of the job for garbage collection and reconcilation.
+	if err := ctrl.SetControllerReference(js, job, r.Scheme); err != nil {
+		return err
+	}
+
+	// Create the job with an exponential back off policy if it already exists.
+	return wait.ExponentialBackoff(backoff, func() (bool, error) {
+		if err := r.Create(ctx, job); err != nil {
+			return false, err
+		}
+		return true, nil
+	})
```

Comment:
We shouldn't do the exponential backoff explicitly, we should just try to force another reconcile based on exponential backoff.

I actually found that controller-manager does that by default when we return an error: https://github.com/kubernetes-sigs/controller-runtime/blob/4000e996a202917ad7d40f02ed8a2079a9ce25e9/pkg/reconcile/reconcile.go#L95

so we shouldn't change anything here

Reply:
Ah interesting, I didn't know that. Done.